central/complianceoperator/v2/checkresults/service/service_impl.go

```diff
@@ -372,8 +444,36 @@ func (s *serviceImpl) GetComplianceProfileCheckDetails(ctx context.Context, requ
 	if err != nil {
 		return nil, errors.Wrapf(errox.InvalidArgs, "Unable to retrieve compliance scan results for query %v", parsedQuery)
 	}
+	if len(scanResults) == 0 {
+		return nil, nil
+	}
+
+	rules, err := s.ruleDS.SearchRules(ctx, search.NewQueryBuilder().AddExactMatches(search.ComplianceOperatorRuleRef, scanResults[0].GetRuleRefId()).ProtoQuery())
+	if err != nil {
+		return nil, errors.Wrapf(errox.InvalidArgs, "Unable to retrieve compliance rule for query %v", parsedQuery)
+	}
+	if len(rules) != 1 {
+		return nil, errors.Wrapf(errox.InvalidArgs, "Unable to process compliance rule for query %v", parsedQuery)
+	}
 
-	return storagetov2.ComplianceV2SpecificCheckResult(scanResults, request.GetCheckName()), nil
+	scanRefQuery := search.NewQueryBuilder().AddExactMatches(search.ComplianceOperatorScanRef, scanResults[0].GetScanRefId()).
+		ProtoQuery()
+	profiles, err := s.profileDS.SearchProfiles(ctx, scanRefQuery)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to retrieve profiles for result %v", parsedQuery)
+	}
+	var convertedControls []*v2.ComplianceControl
+	if len(profiles) != 0 {
+		// TODO(ROX-22362): implement tailored profiles
+		log.Warnf("Unable to find profiles for result %v.  It is possible results match a tailored profile which have not been implemented in Compliance V2", parsedQuery)
+		controls, err := utils.GetControlsForScanResults(ctx, s.ruleDS, []string{rules[0].GetName()}, profiles[0].GetName(), s.benchmarkDS)
+		if err != nil {
+			return nil, errors.Wrapf(errox.InvalidArgs, "Unable to retrieve controls for compliance scan results %v", parsedQuery)
+		}
```

Comment:
Maybe I'm not understanding the code here but to me the warning indicates that no profiles were found when the if checks that at least one was found. Did you mean?
```suggestion
	if len(profiles) == 0 {
		// TODO(ROX-22362): implement tailored profiles
		log.Warnf("Unable to find profiles for result %v.  It is possible results match a tailored profile which have not been implemented in Compliance V2", parsedQuery)
		} else {
		controls, err := utils.GetControlsForScanResults(ctx, s.ruleDS, []string{rules[0].GetName()}, profiles[0].GetName(), s.benchmarkDS)
		if err != nil {
			return nil, errors.Wrapf(errox.InvalidArgs, "Unable to retrieve controls for compliance scan results %v", parsedQuery)
		}
```